system/jlib/jmutex.hpp

```diff
@@ -410,8 +410,23 @@ class jlib_decl  SpinLock
             nesting++;
             return;
         }
-        while (!atomic_cas(&value,1,0)) 
-            ThreadYield(); 
+
+        unsigned i = 0;
+        const unsigned maxSpins = 100;
```

Comment:
In standalone benchmark testing a value of 20 for maxSpins was better for all cases.